gulpfile.js

```diff
@@ -0,0 +1,164 @@
+  }
+}
+
+async function e2eInstall() {
+  execSync('npm install', {stdio: 'inherit', cwd: TEST_DIR});
+}
+
+function e2eEnsureConfigFile() {
+  if (!fs.existsSync(CONFIG)) {
+    // copy the template to the local config location
+    return src(CONFIG_TEMPLATE)
+      .pipe(rename(CONFIG_NAME))
+      .pipe(dest(CONFIG_DIR));
+  }
+  return Promise.resolve();
+}
+
+
+function promptTestSuite() {
+  return testSuites[
+    readlineSync.keyInSelect(testSuites, 'Select a test suite: ', {cancel: false})
+  ];
+}
+
+function promptOsioUrl(currentValue) {
+  let cancel = false;
+  // add the current URL as a choice in case it's custom
+  if (currentValue.length > 0) {
+    cancel = `use current URL (${currentValue})`;
+  }
+
+  // create the array of url choices
+  const urls = Object.keys(testUrls).map(
+    (name) => `${name}${testUrls[name] ? ` (${testUrls[name]})` : ''}`,
+  );
+  const choice = readlineSync.keyInSelect(urls, 'Select a test URL: ', {cancel});
+
+  // if custom, ask for input
+  if (choice === 0) {
+    return readlineSync.question(`Enter a custom test URL: `);
+  } else if (choice !== -1) {
+    return testUrls[Object.keys(testUrls)[choice]];
+  }
+  return currentValue;
+}
+
+function promptInput(field) {
+  return readlineSync.question(`Enter value for ${field}: `, {
+    hideEchoBack: field.toLowerCase().indexOf('password') !== -1,
+  });
+}
+
+function promptResetEnvironment(field) {
+  return readlineSync.keyInYN(`Reset environment after tests (WARNING: this is desctructive to your account. If unsure choose NO): `);
+}
+
+function promptFeatureLevel() {
+  return featureLevels[
+    readlineSync.keyInSelect(featureLevels, 'Select your OSIO account feature level: ', {cancel: false})
+  ];
+}
+
+function e2eEditConfig() {
+  // read the existing config and
+  return src(CONFIG)
+    .pipe(
+      replace(/\${([a-zA-Z-_]+):-(.*?)}/g, (match, p1, p2) => {
+        let value = null;
+        switch (p1) {
+          case 'TEST_SUITE':
+            value = promptTestSuite();
+            break;
+          case 'RESET_ENVIRONMENT':
+            value = promptResetEnvironment();
+            break;
+          case 'OSIO_URL':
+            value = promptOsioUrl(p2);
+            break;
+          case 'FEATURE_LEVEL':
+            value = promptFeatureLevel();
+            break;
+          default:
+            if (p2 === '') {
+              value = promptInput(p1);
+            }
+        }
+        return `\${${p1}:-${value == null ? p2 : value}}`;
+      }),
+    )
+    .pipe(dest(CONFIG_DIR));
+}
+
+function e2eClearConfig() {
+  return src(CONFIG, {read: false, allowEmpty: true}).pipe(clean({force: true}))
+}
+
+async function e2eStart() {
+  try {
+    execSync('npm start', {stdio: 'inherit', cwd: TEST_DIR});
+  } catch (err) {
+    // suppress as the test would have already logged to console
+  }
+}
+
+const e2eConfig = series(e2eEnsureConfigFile, e2eEditConfig);
```

Comment:
Would it make sense to put `e2eConfig` and `e2eReconfig` directly to `fabric8-test` repository? WDYT @ljelinkova?